Upgrade bundled Google Test version

## Description
Google Test version downloaded by rnp build scripts fails to build with GCC 11
```
[ 11%] Building CXX object src/lib/CMakeFiles/librnp-obj.dir/__/librepgp/stream-write.cpp.o
In file included from /home/odsk/Ribose/rnp/build/_deps/googletest-src/googletest/src/gtest-all.cc:42:
/home/odsk/Ribose/rnp/build/_deps/googletest-src/googletest/src/gtest-death-test.cc: In function ‘bool testing::internal::StackGrowsDown()’:
/home/odsk/Ribose/rnp/build/_deps/googletest-src/googletest/src/gtest-death-test.cc:1301:24: error: ‘dummy’ may be used uninitialized [-Werror=maybe-uninitialized]
 1301 |   StackLowerThanAddress(&dummy, &result);
      |   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
/home/odsk/Ribose/rnp/build/_deps/googletest-src/googletest/src/gtest-death-test.cc:1290:13: note: by argument 1 of type ‘const void*’ to ‘void testing::internal::StackLowerThanAddress(const void*, bool*)’ declared here
 1290 | static void StackLowerThanAddress(const void* ptr, bool* result) {
      |             ^~~~~~~~~~~~~~~~~~~~~
/home/odsk/Ribose/rnp/build/_deps/googletest-src/googletest/src/gtest-death-test.cc:1299:7: note: ‘dummy’ declared here
 1299 |   int dummy;
      |       ^~~~~

```
This is a known bug, already fixed in upstream
https://github.com/google/googletest/issues/3219
## Steps to Reproduce

1. Configure build with options `-DBUILD_TESTING=on -DCMAKE_BUILD_TYPE=Debug -DDOWNLOAD_GTEST=On`
2. Use GCC 11 to build 
```
gcc --version
gcc (Gentoo 11.2.0 p1) 11.2.0
```
3. try to build

--- a/src/tests/CMakeLists.txt
+++ b/src/tests/CMakeLists.txt
@@ -47,11 +47,16 @@ elseif (NOT DOWNLOAD_GTEST)
   find_package(GTest REQUIRED)
   set(GTestMain GTest::Main)
 else()
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS_EQUAL "4.8.5")
+    set(GTEST_GIT_TAG "c43f710")
+  else()
+    set(GTEST_GIT_TAG "HEAD")
+  endif()
   # download and build googletest
   include(FetchContent)
   FetchContent_Declare(googletest
     GIT_REPOSITORY  https://github.com/google/googletest.git
-    GIT_TAG         c43f710
+    GIT_TAG         "${GTEST_GIT_TAG}"
   )
   # maintain compiler/linker settings on Windows
   set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)

